packages/react/src/components/Tree/TreeItem.tsx

```diff
@@ -106,28 +105,43 @@ class TreeItem extends UIComponent<WithAsProp<TreeItemProps>> {
     accessibility: treeItemBehavior,
   }
 
-  titleRef = React.createRef<HTMLElement>()
+  itemRef = React.createRef<HTMLElement>()
   treeRef = React.createRef<HTMLElement>()
 
   actionHandlers = {
-    getFocusFromParent: e => {
-      const { open } = this.props
-      if (open) {
+    performClick: e => {
+      e.preventDefault()
+      e.stopPropagation()
+
+      _.invoke(this.props, 'onTitleClick', e, this.props)
+    },
+    collapseOrReceiveFocus: e => {
+      const { items, open } = this.props
+
+      e.preventDefault()
+
+      // Focuses the title if the event comes from a child item.
+      if (e.currentTarget !== e.target && items && items.length) {
+        e.stopPropagation()
+        this.itemRef.current.focus()
+      } else if (open) {
         e.stopPropagation()
-        this.titleRef.current.focus()
+        _.invoke(this.props, 'onTitleClick', e, this.props)
       }
     },
-    setFocusToFirstChild: e => {
+    expandOrPassFocus: e => {
```

Comment:
nit: `expandOrPassFocusToSubtree` - wouldn't be more meanigful?

Reply:
same sort of comment - as a client that is consuming this API, I don't understand what `passFocus` means. Would suggest shorter form of what @sophieH29 has suggested: `expandOrFocusSubtree`